src/worker.js

```diff
@@ -211,14 +213,23 @@ self.onmessage = (e) => {
       Module['PThread'].receiveObjectTransfer(e.data);
       Module['PThread'].threadInit();
 
-#if EMBIND
-      // Embind must initialize itself on all threads, as it generates support JS.
-      // We only do this once per worker since they get reused
       if (!initializedJS) {
+#if EMBIND
+        // Embind must initialize itself on all threads, as it generates support JS.
+        // We only do this once per worker since they get reused
         Module['___embind_register_native_and_builtin_types']();
+#endif // EMBIND
+
+        // Execute any proxied work that came in before the thread was
+        // initialized. Only do this once because it is only possible for
+        // proxying notifications to arrive before thread initialization on
+        // fresh workers.
+        for (const queue of pendingNotifiedProxyingQueues) {
```

Comment:
Maybe just `pendingNotifiedProxyingQueues.values().forEach()` actually since we haven't actually started allowing let/const in the codebase yet.

Reply:
Oh, I didn't realize that. Will fix.

Reply:
Although there are already lots of uses of both `let` and `const`, so this doesn't seem worth fixing.

Reply:
I don't think there are any in the generated code yet..  are you sure?

Reply:
Yes, there is a lot of `const` in library_webgl.js and `let` in multiple libraries including library.js, library_wasm_worker.js, and library_webgl.js.

Reply:
Hmm.. maybe I misspoke.   At least we are not using ` .. of ..` yet.

Reply:
https://github.com/emscripten-core/emscripten/blob/main/src/library_webgl.js#L578 !

Reply:
Looks like that was introduced fairly recently :(     But in general we don't support it yet.

Reply:
(Was introduced in #16189)

Reply:
Looks like I use it in `library_wasmfs_node.js` as well. How about a follow-up PR to fix all these uses?

Reply:
You mean introduce an extra usage of the `of` operator, only to remove it?  Or you mean remove that `of` operator from this PR and then fix the others?   I don't really mind what order but its seem odd to land something you plan to remove.

Reply:
I just don't want to have to wait for CI to run again :cry: